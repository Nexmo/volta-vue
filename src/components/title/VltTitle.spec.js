import { shallowMount } from '@vue/test-utils';
import { VltTitle } from '..';

describe('vlt-title', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(VltTitle, {
      propsData: {
        icon: 'rocket',
        title: 'Test Title',
      },
    });
  });

  test('renders correctly with text "Test Title" and rocket icon', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('icon is red', async () => {
    wrapper.setProps({ color: 'red' });
    await wrapper.vm.$nextTick();
    expect(wrapper.element).toMatchSnapshot();
  });
});
