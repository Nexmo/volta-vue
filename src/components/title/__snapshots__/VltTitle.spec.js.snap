// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vlt-title icon is red 1`] = `
<h2
  class="Vlt-title--icon"
>
  <vlt-icon-stub
    clickable="true"
    color="red"
    icon="rocket"
    noclass="false"
    pointer="false"
    small="false"
    smaller="false"
  />
  Test Title
</h2>
`;

exports[`vlt-title renders correctly with text "Test Title" and rocket icon 1`] = `
<h2
  class="Vlt-title--icon"
>
  <vlt-icon-stub
    clickable="true"
    color=""
    icon="rocket"
    noclass="false"
    pointer="false"
    small="false"
    smaller="false"
  />
  Test Title
</h2>
`;
